Add Collection Controller Documentation (#37)
kuzzleio/sdk-csharp: doc/1/controllers/collection/list/index.md

---
code: true
type: page
title: ListAsync
description: Returns the collection list of an index.
---

# ListAsync

Returns the list of collections associated to a provided index.

The returned list is sorted in alphanumerical order.
The `from` and `size` arguments allow pagination. They are returned in the response if provided.

## Arguments

```csharp
public async Task<JObject> ListAsync(
      string index,
      int? from = null,
      int? size = null,

>       string type = null

Reviewer: If the default value is "all", shouldn't this be `string type = "all"`?

(also shouldn't we create an enum for that one?)

---
code: true
type: page
title: ListAsync
description: Returns the collection list of an index.
---

# ListAsync

> Returns the complete list of realtime and stored collections in requested index sorted by name in alphanumerical order.

Reviewer: ```suggestion
Returns the list of collections associated to a provided index.

The returned list is sorted in alphanumerical order.
```

The `from` and `size` arguments allow pagination. They are returned in the response if provided.

## Arguments

```csharp
public async Task<JObject> ListAsync(
      string index,
      int? from = null,
      int? size = null,
      string type = null
    );
```

| Argument | Type              | Description |
|----------|-------------------|-------------|
| `index`  | <pre>string</pre> | Index name  |

> ### options

Reviewer: Inconsistent with our documentation template: optional parameters in the C# SDK should be put in the arguments table.

Additional query options

| Property | Type<br/>(default)            | Description                                                                                         |
|----------|-------------------------------|-----------------------------------------------------------------------------------------------------|
| `from`   | <pre>int?</pre><br>(`null`)   | Offset of the first result                                                                          |
| `size`   | <pre>int?</pre><br>(`null`)   | Maximum number of returned results                                                                  |

> | `type`   | <pre>string</pre><br>(`null`) | filters the returned collections. Allowed values: `all`, `stored` and `realtime` (default : `all`). |

Reviewer 2: ```suggestion
| `type`   | <pre>string</pre><br>(`null`) | Filters the returned collections. Allowed values: `all`, `stored` and `realtime` (default : `all`). |
```
